sdk/src/store.rs

```diff
@@ -2372,34 +2385,44 @@ impl Store {
         // replace the source with correct asset hashes so that the claim hash will be correct
         if needs_hashing {
             let pc = self.provenance_claim_mut().ok_or(Error::ClaimEncoding)?;
-
             // get the final hash ranges, but not for update manifests
             intermediate_stream.rewind()?;
             output_stream.rewind()?;
             std::io::copy(output_stream, &mut intermediate_stream)?; // can remove this once we can get a CAIReader from CAIReadWrite safely
-            let mut new_hash_ranges =
-                object_locations_from_stream(format, &mut intermediate_stream)?;
-            let updated_hashes = if pc.update_manifest() {
-                Vec::new()
-            } else {
-                Store::generate_data_hashes_for_stream(
-                    &mut intermediate_stream,
-                    pc.alg(),
-                    &mut new_hash_ranges,
-                    true,
-                )?
-            };
 
-            // patch existing claim hash with updated data
-            for hash in updated_hashes {
-                pc.update_data_hash(hash)?;
+            if let Some(handler) = get_assetio_handler(format) {
+                // If our asset supports box hashing, generate and update
+                // the existing box hash assertion.
+                if let Some(box_hash_handler) = handler.asset_box_hash_ref() {
```

Comment:
The replace block should not be needed for the box hash case if you calculate it as stated above.